internals/router/authmanagement/authentification_mode.go

```diff
@@ -0,0 +1,52 @@
+package authmanagement
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/spf13/viper"
+)
+
+type AuthenticationMode struct {
+	Mode string `json:"mode"`
+}
+
+type AuthenticationModeQuerier struct{}
+
+var (
+	_globalQuerierMu sync.RWMutex
+	_globalQuerier   *AuthenticationModeQuerier
+)
+
+func New() *AuthenticationModeQuerier {
+	querier := &AuthenticationModeQuerier{}
+	return querier
+}
+
+func S() *AuthenticationModeQuerier {
+	_globalQuerierMu.RLock()
+	defer _globalQuerierMu.RUnlock()
+
+	return _globalQuerier
+}
+
+func ReplaceGlobals(querier *AuthenticationModeQuerier) func() {
+	_globalQuerierMu.Lock()
+	defer _globalQuerierMu.Unlock()
+
+	prev := _globalQuerier
+	_globalQuerier = querier
+
+	return func() {
+		ReplaceGlobals(prev)
+	}
+}
+
+func (querier AuthenticationModeQuerier) GetMode() (AuthenticationMode, error) {
+
+	mode := viper.GetString("AUTHENTICATION_MODE")
+	if mode == "" {
```

Comment:
mode == "" then mode = "BASIC" ? maybe better than a error to have basic as default

Reply:
es, instead of returning an error, I can default to BASIC authentication. However, this situation should not occur since the application should execute with at least one type of authentication. But I can make that adjustment, yes.